src/gebt_poc/gen_alpha_2D.cpp

```diff
@@ -0,0 +1,334 @@
+            this->is_converged_ = true;
+            break;
+        }
+
+        auto iteration_matrix = linearization_parameters->IterationMatrix(
+            h, kBetaPrime, kGammaPrime, gen_coords_next, delta_gen_coords, velocity_next,
+            acceleration_next, lagrange_mults_next
+        );
+
+        if (this->is_preconditioned_) {
+            iteration_matrix = gen_alpha_solver::multiply_matrix_with_matrix(iteration_matrix, dr);
+            iteration_matrix = gen_alpha_solver::multiply_matrix_with_matrix(dl, iteration_matrix);
+
+            Kokkos::parallel_for(
+                size,
+                KOKKOS_LAMBDA(const size_t i) { residuals(i) = residuals(i) * kBetaLocal * h * h; }
+            );
+        }
+
+        auto soln_increments = Kokkos::View<double*>("soln_increments", residuals.size());
+        Kokkos::deep_copy(soln_increments, residuals);
+        gen_alpha_solver::solve_linear_system(iteration_matrix, soln_increments);
+
+        Kokkos::View<double*> delta_x("delta_x", delta_gen_coords.size());
+        Kokkos::parallel_for(
+            delta_gen_coords.size(),
+            // Take negative of the solution increments to update generalized coordinates
+            KOKKOS_LAMBDA(const size_t i) { delta_x(i) = -soln_increments(i); }
+        );
+
+        if (n_constraints > 0) {
+            Kokkos::View<double*> delta_lagrange_mults("delta_lagrange_mults", n_constraints);
+
+            if (this->is_preconditioned_) {
+                Kokkos::parallel_for(
+                    n_constraints,
+                    KOKKOS_LAMBDA(const size_t i) {
+                        // Take negative of the solution increments to update Lagrange
+                        // multipliers
+                        delta_lagrange_mults(i) =
+                            -soln_increments(i + delta_gen_coords.size()) / (kBetaLocal * h * h);
+                        lagrange_mults_next(i) += delta_lagrange_mults(i);
+                    }
+                );
+            } else {
+                Kokkos::parallel_for(
+                    n_constraints,
+                    KOKKOS_LAMBDA(const size_t i) {
+                        // Take negative of the solution increments to update Lagrange
+                        // multipliers
+                        delta_lagrange_mults(i) = -soln_increments(i + delta_gen_coords.size());
+                        lagrange_mults_next(i) += delta_lagrange_mults(i);
+                    }
+                );
+            }
+        }
+
+        for (size_t node = 0; node < n_nodes; node++) {
+            // Update the velocity, acceleration, and constraints based on the increments
+            Kokkos::parallel_for(
+                size,
+                KOKKOS_LAMBDA(const size_t i) {
+                    delta_gen_coords(node, i) += delta_x(i) / h;
+                    velocity_next(node, i) += kGammaPrime * delta_x(i);
+                    acceleration_next(node, i) += kBetaPrime * delta_x(i);
```

Comment:
@faisal-bhuiyan 
Is this correct?  `size` is set to `velocity.size()` which is the same as `n_nodes * kNumberOfLieAlgebraComponents`.  

Instead, line 231 should be over `kNumberOfLieAlgebraComponents`, and you need to do the correct index math to access the elements of `delta_x`.  

Am I missing something here?

Reply:
`velocity.size()` = 6 in this context. However, I fixed all the index related issues in my local branch and planning to push these changes once we get through this and likely a couple other PRs. Does that sound good?

Reply:
As long as the results are correct, sounds good.